app/src/main/java/com/github/se/polyfit/data/remote/firebase/PostFirebaseRepository.kt

```diff
@@ -187,6 +202,7 @@ class PostFirebaseRepository(
 
                 completedBatches++
                 if (completedBatches == batches.size) {
+                  previousListOfPosts = posts
                   completion(posts)
                 }
```

Comment:
I have read in a few sources such as these :
https://kotlinlang.org/docs/shared-mutable-state-and-concurrency.html#mutual-exclusion
https://jacquessmuts.github.io/post/coroutine_sync_mutex/

That within a coroutine scope the use of synchronization is better done with a `MutexLock` paired with `withContext()` as issues could arise in the Coroutine, however considering this is a V1 and the code currently doesn't seem to be having any issues I think it should be fine to let it as it is. Just suggesting for later uses

Reply:
this is the code of a previous PR